src/main/java/seedu/address/logic/commands/eventvolunteercommands/EventAddVolunteerCommand.java

```diff
@@ -63,11 +68,40 @@ public CommandResult execute(Model model) throws CommandException {
         if (eventToAssign.hasVolunteer(volunteerToAssign)) {
             throw new CommandException(MESSAGE_DUPLICATE_VOLUNTEER);
         }
+
+        Set<EventName> assignedEvents = volunteerToAssign.getAssignedEvents();
+        for (EventName eventName : assignedEvents) {
+            Predicate<Event> eventPredicate = e -> e.getEventName().equals(eventName);
+            Event otherEvent = model.getEventStorage().getEvent(eventName);
+            if (hasClashingEvents(eventToAssign, otherEvent)) {
+                throw new CommandException(MESSAGE_CLASHING_EVENTS);
+            }
+        }
         volunteerToAssign.addEvent(eventToAssign);
         eventToAssign.addVolunteer(volunteerToAssign);
         return new CommandResult(String.format(MESSAGE_SUCCESS, Messages.format(eventToAssign),
                 eventToAssign.getAssignedVolunteers().size()));
     }
+
+    /**
+     * Checks if {@code eventToAssign} clashes with {@code otherEvent}
+     * @param eventToAssign The Event to be assigned.
+     * @param otherEvent The Event to be checked.
+     * @return true if the two events clash.
+     */
+    public boolean hasClashingEvents(Event eventToAssign, Event otherEvent) {
+        boolean startDateTimeClashes = eventToAssign.getStartDate().dateAndTime
+                .isAfter(otherEvent.getStartDate().dateAndTime)
+                && eventToAssign.getStartDate().dateAndTime.isBefore(otherEvent.getEndDate().dateAndTime);
+        boolean endDateTimeClashes = eventToAssign.getEndDate().dateAndTime
+                .isAfter(otherEvent.getStartDate().dateAndTime)
+                && eventToAssign.getEndDate().dateAndTime.isBefore(otherEvent.getEndDate().dateAndTime);
+        boolean sameStartAndEndDateTime = eventToAssign.getStartDate().dateAndTime
+                .isEqual(otherEvent.getStartDate().dateAndTime)
+                && eventToAssign.getEndDate().dateAndTime.isEqual(otherEvent.getEndDate().dateAndTime);
```

Comment:
not fully certain about this. this encompasses only one case where the event start and end are both the same, which is pretty rare